### PR TITLE
[doc][vulkan] Merge vulkan_and_spirv.md into other docs

### PR DESCRIPTION
**Pull request by antiagainst**
This commit moves the contents in vulkan_and_spirv.md into other
suitable places and retires it.

### CODE REVIEW

File: docs/GetStarted/generic_vulkan_env_setup.md [hunk 1]
```diff
@@ -95,21 +95,61 @@ $ LD_LIBRARY_PATH={PATH_TO_VULKAN_SDK}/x86_64/lib/
...
+    example, You can force enable `VK_LAYER_LUNARG_api_dump` to have a detailed
+    dump of all Vulkan API calls made by the application. You can force enable
+    `VK_LAYER_LUNARG_core_validation` to validate the API calls made by the
+    application.
 *   `VK_LAYER_PATH`: override the loader's standard layer libary search folders.
 
 Please see the [Vulkan loader's documentation][VulkanLoaderEnvVars] for detailed
 explanation for these variables.
 
+### Setting environmet variables for Bazel test
```
Comment (ScottTodd) [inline docs/GetStarted/generic_vulkan_env_setup.md:129]:
```suggestion
### Setting environment variables for Bazel tests
```

File: docs/GetStarted/generic_vulkan_env_setup.md [hunk 2]
```diff
@@ -95,21 +95,61 @@ $ LD_LIBRARY_PATH={PATH_TO_VULKAN_SDK}/x86_64/lib/
...
+
+Bazel runs tests in a sandbox and environment variables must be passed through
+to the test runner. Consider putting environment setup in a `user.bazelrc` to
+trouble shooting a test run by Bazel. For example:
```
Comment (ScottTodd) [inline docs/GetStarted/generic_vulkan_env_setup.md:133]:
```suggestion
to the test runner. Consider putting environment setup in a `user.bazelrc`. For
example:
```

Keep the original language or adjust. When troubleshooting, you would just set the arguments directly on the command line, you persist common configurations to bazelrc files to save typing.

Developers are also free to add their own configurations, like

```
test:vkswiftshader --test_env="LD_LIBRARY_PATH=...
test:vkswiftshader --test_env="VK_LAYER_PATH=...
test:vknative --test_env="LD_LIBRARY_PATH=...
test:vknative --test_env="VK_LAYER_PATH=...
```

then use with `bazel test --config=vkswiftshader ...`

(actually now that I'm typing that out I might do that setup on my Linux machine... I've been saving bash history that sets all the flags :P)

Comment (antiagainst) [reply docs/GetStarted/generic_vulkan_env_setup.md]: Good points! I've updated suitable places to have this handy settings.

File: docs/GetStarted/generic_vulkan_env_setup.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ The Vulkan loader sits between the Vulkan application, which calls Vulkan APIs,
 and the ICDs, which implements these Vulkan APIs. Vulkan layers agument the
 Vulkan system to provide optional features like validation and debugging. The
 Vulkan loader composes a chain of requested layers, which processes the Vulkan
-application's API calls one by one and finallly redirects the API calls made by
+application's API calls one by one, and finallly redirects the API calls made by
```
Comment (ScottTodd) [inline docs/GetStarted/generic_vulkan_env_setup.md:27]:
```suggestion
application's API calls one by one, and finally redirects the API calls made by
```
(typo)

**Pull request closed without merging**